packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js

```diff
@@ -2,17 +2,99 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { useCallback } from '@wordpress/element';
+import { useCallback, useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore, useEntityBlockEditor } from '@wordpress/core-data';
+import { BlockEditorProvider, Inserter } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import SidebarNavigationScreen from '../sidebar-navigation-screen';
-import NavigationInspector from '../navigation-inspector';
 import { useHistory } from '../routes';
+import NavigationMenuContent from './navigation-menu-content';
+import SidebarButton from '../sidebar-button';
+
+const NAVIGATION_MENUS_QUERY = { per_page: -1, status: 'publish' };
+const settings = {
+	allowedBlockTypes: [
+		'core/navigation-link',
```

Comment:
The navigation link is an allowed block but some blocks like "Custom Link" and "Post Link" don't appear on the inserter and is impossible to add them:
<img width="401" alt="Screenshot 2023-03-02 at 23 19 58" src="https://user-images.githubusercontent.com/11271197/222586215-db93e3f8-b372-4e00-9d65-74d222372d43.png">
 
 I'm not sure yet why they don't appear 🤔

Reply:
Is this also a problem in trunk?

Reply:
I've explained the reasons for this here https://github.com/WordPress/gutenberg/pull/48689#issuecomment-1452092236 

I'm not sure why it doesn't happen in trunk but I'll explore a fix for this

Reply:
Ok, I've fixed the issue. There's one small thing remaining though. You can see when loading the page that the "navigation menu" appears "empty" for a second. I didn't find the right check to add to maintain the `isLoading` state properly until all the inner blocks are loaded. Maybe @scruffian has an idea on what we can check here.

Reply:
I found a small hack to load for the navigation block to load its inner blocks but I still see like a couple milliseconds where the navigation menu shows "an empty state". It's not perfect but maybe it's good enough to start.